ocaml/file_formats/cms_format.mli

```diff
@@ -0,0 +1,52 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                   Fabrice Le Fessant, INRIA Saclay                     *)
+(*                                                                        *)
+(*   Copyright 2012 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** cms and cmsi files format. *)
+
+type cms_infos = {
+  cms_modname : Compilation_unit.t;
+  cms_comments : (string * Location.t) list;
+  cms_sourcefile : string option;
+  cms_builddir : string;
+  cms_source_digest : string option;
+  cms_uid_to_loc : Location.t Shape.Uid.Tbl.t;
+  cms_uid_to_attributes : Parsetree.attributes Shape.Uid.Tbl.t;
+  cms_impl_shape : Shape.t option; (* None for mli *)
+}
+
+type error =
+  Not_a_shape of string
+
+exception Error of error
+
+(** [read filename] opens filename, and extract the cms_infos. It
+    can be used with .cmi, .cms and .cmsi files.
```

Comment:
I'm not sure why this comment says the function can be used on `.cmi` files, it seems to be that it will fail on those files?

Reply:
Forgot to update the comment after cleaning up the file from a previous revision. Sorry for this!